### PR TITLE
hotfix/JM-8077: Deferral Maintenance Screen Not Sorting on Pagination

### PR DESCRIPTION
**Pull request by callum-r-young**
### JIRA link (if applicable) ###

https://centralgovernmentcgi.atlassian.net/browse/JM-8077

### Change description ###

Updated deferral maintenance sorting to persist through filtering and pagination.

Updated deferral maintenance checking to persist through pagination, and select all to select all when filtered.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: server/routes/pool-management/deferral-maintenance/deferral-maintenance.controller.js [hunk 1]
```diff
@@ -178,15 +185,26 @@
...
+              return sortOrder === 'ascending' ? -1 : 1;
+            }
+            if (sortA > sortB) {
+              return sortOrder === 'ascending' ? 1 : -1;
+            }
 
-      if (currentPage > 1) {
-        offset = modUtils.constants.PAGE_SIZE * (currentPage - 1);
+            return 0;
+          });
```
Comment (ricardofreitasrocha) [inline server/routes/pool-management/deferral-maintenance/deferral-maintenance.controller.js:201]: Could we use a `localeCompare()` on this sort function?
could probably write something like

```js
data.deferrals.sort((a, b) => {
  if (ascending) a.toUpperCase().localCompare(b.toUpperCase())
  else .... the opposite
})
```

File: server/routes/pool-management/deferral-maintenance/deferral-maintenance.controller.js [hunk 2]
```diff
@@ -208,6 +234,8 @@
       var juror, total;
 
       if (req.params.jurorNumber === 'all') {
+        console.log(req.session.deferralMaintenance.filtered);
+        console.log(req.session.deferralMaintenance.filtered.length)
```
Comment (ricardofreitasrocha) [inline server/routes/pool-management/deferral-maintenance/deferral-maintenance.controller.js:238]: remove

**Review by ricardofreitasrocha — CHANGES_REQUESTED** [review]

File: server/routes/pool-management/deferral-maintenance/deferral-maintenance.controller.js (hunk 2 above)
Comment (callum-r-young) [reply server/routes/pool-management/deferral-maintenance/deferral-maintenance.controller.js]: removed.

File: server/routes/pool-management/deferral-maintenance/deferral-maintenance.controller.js (hunk 1 above)
Comment (callum-r-young) [reply server/routes/pool-management/deferral-maintenance/deferral-maintenance.controller.js]: Updated to use localeCompare() :)

**Review by ricardofreitasrocha — APPROVED** [review]

**Pull request merged**